pkg/whitesource/whitesource.go

```diff
@@ -186,56 +172,33 @@ func (s *System) GetProjectVitals(projectToken string) (*Project, error) {
 		ProjectToken: projectToken,
 	}
 
-	respBody, err := s.sendRequest(req)
+	err := s.sendRequestAndDecodeJSON(req, &wsResponse)
 	if err != nil {
-		return nil, errors.Wrap(err, "WhiteSource request failed")
+		return Project{}, errors.Wrap(err, "WhiteSource request failed")
 	}
 
-	err = json.Unmarshal(respBody, &wsResponse)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to parse WhiteSource response")
+	if len(wsResponse.ProjectVitals) == 0 {
+		return Project{}, errors.Wrapf(err, "no project with token '%s' found in WhiteSource", projectToken)
 	}
 
-	return &wsResponse.ProjectVitals[0], nil
+	return wsResponse.ProjectVitals[0], nil
 }
 
-// GetProjectByName returns the finds and returns a project by name
-func (s *System) GetProjectByName(productToken, projectName string) (*Project, error) {
-	var project *Project
+// GetProjectByName fetches all projects and returns the one matching the given projectName, or none, if not found
+func (s *System) GetProjectByName(productToken, projectName string) (Project, error) {
```

Comment:
incompatible change of API

Reply:
I think it is worth it. Why have a mix of pointer versus value returns in the same API?

Reply:
ok, but why then not moving pointers  :-)

Reply:
I don't remember, but I think I went with the style of the majority of the remaining functions...